lib/sshkit/backends/netssh.rb

```diff
@@ -123,7 +129,12 @@ def transfer_summarizer(action, options = {})
       end
 
       def execute_command(cmd)
-        output.log_command_start(cmd)
+
+        # Redaction
+        cmd_w_redaction = cmd.dup
+        cmd.dup.args.map!{|arg| arg.is_a?(Array) ? arg[0] : arg }
```

Comment:
This does redact the arguments from the logs (as verified by the unit test), but I suspect it is also redacting (and thus breaking) the actual command that is being executed.

The issue is that `dup` only makes a shallow copy. Therefore `cmd.dup.args` is the same reference as `cmd.args`, and so `args.map!` is mutating both copies of the command. Furthermore, don't we want to log `arg[0]` (i.e. the `*REDACTED*` string), but execute `arg[1]`? I don't see where the latter is taking place in the code.

I tested this branch using a Capistrano task:

```ruby
task :test_redact do
  on roles(:all) do
    execute :echo, redact("secret!"), "> test_redact"
    puts capture(:cat, "test_redact") #  this should print `secret!`
  end
end
```

And confirmed that `secret!` is being redacted from the log, but the original `secret!` is never actually passed to the command being executed:

```
00:00 test_redact
      01 echo *REDACTED* > test_redact
    ✔ 01 deployer@xx.xx.xx.xx 2.344s
*REDACTED*
```

Reply:
Hmm, ok, I’ll run more tests. I had manually confirmed it was working but could have just confused myself into thinking it was fine